sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/access/SormasToSormasDiscoveryService.java

```diff
@@ -119,24 +119,24 @@ public SormasServerDescriptor getSormasServerDescriptorById(String id) {
 		}
 
 		try {
-			RedisCommands<String, String> redis = createRedisConnection();
-			if (redis == null) {
+			KvClient etcd = createEtcdClient();
+			if (etcd == null) {
 				LOGGER.error((I18nProperties.getString(Strings.errorSormasToSormasServerAccess)));
 				return Collections.emptyList();
 			}
 
-			List<String> keys = redis.keys(String.format(sormasToSormasConfig.getKeyPrefixTemplate(), "*"));
-
-			// remove own Id from the set
-			keys.remove(String.format(sormasToSormasConfig.getKeyPrefixTemplate(), sormasToSormasConfig.getId()));
+			final String keyPrefix = String.format(sormasToSormasConfig.getKeyPrefixTemplate(), "");
+			return etcd.get(ByteString.copyFromUtf8(keyPrefix))
+				.asPrefix()
+				.sync()
+				.getKvsList()
+				.stream()
+				.filter(
+					// this ensures that the own key (i.e., /s2s/$instance_id) is removed from the list
+					e -> !e.getKey().toStringUtf8().equals(String.format(sormasToSormasConfig.getKeyPrefixTemplate(), sormasToSormasConfig.getId())))
```

Comment:
`String.format(sormasToSormasConfig.getKeyPrefixTemplate(), sormasToSormasConfig.getId())`  can be done outside the filter as it's always the same

Reply:
yes done :)